workspaces/ballerina/ballerina-visualizer/src/views/TypeDiagram/index.tsx

```diff
@@ -399,6 +536,70 @@ export function TypeDiagram(props: TypeDiagramProps) {
                 </ViewContent>
             </View>
             {/* Panel for editing and creating types */}
+            <EditorContext.Provider value={{ stack, push: pushTypeStack, pop: popTypeStack, peek: peekTypeStack, replaceTop: replaceFirst }}>
+                {(typeEditorState.editingTypeId || typeEditorState.isTypeCreatorOpen) && typeEditorState.editingType?.codedata?.node !== "CLASS" && (
+                    <PanelContainer
+                        title={typeEditorState.editingTypeId ?
+                            `Edit Type${getTypeKindDisplayName(typeEditorState.editingType?.codedata?.node) ?
+                                ` : ${getTypeKindDisplayName(typeEditorState.editingType?.codedata?.node)}` :
+                                ''}` :
+                            "New Type"
+                        }
+                        show={true}
+                        onClose={onTypeEditorClosed}
```

Comment:
Can we check why the same forms are being reused at L604 as well? The expected behavior is to open the TypeEditor when the user clicks Edit, and in that case, the FormTypeEditor should be rendered.